buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -600,9 +600,10 @@ class BeamModulePlugin implements Plugin<Project> {
                   ? 'apache.releases.https' : 'apache.snapshots.https')
           def m2SettingCreds = new XmlSlurper().parse(settingsXml).servers.server.find { server -> serverId.equals(server.id.text()) }
           if (m2SettingCreds) {
+            def GroovyShell shell = new GroovyShell(new Binding([env:System.getenv()]))
             credentials {
-              username m2SettingCreds.username.text()
-              password m2SettingCreds.password.text()
+              username shell.evaluate('"' + m2SettingCreds.username.text() +'"')
+              password shell.evaluate('"' + m2SettingCreds.password.text() +'"')
```

Comment:
Does this not change the semantics from using the username/password found in the settings to interpreting them as environment variables names and loading them from the environment? If so, shouldn't this be configurable?

Reply:
It doesn't change semantics but expands environment variables, when present.

Reply:
At the least I have to make sure for my password not to contain `$}{` or any Groovy keywords. I think this also allows arbitrary groovy code to be injected into the build code. Could we introduce a flag in the xml settings to indicate whether we have an environment variable or a plain password?

Reply:
What about considering using a plugin to do this, there seems to be a few promising ones:
https://github.com/datlowe/gradle-maven-publish-auth
https://github.com/sebersole/gradle-maven-publish-auth
(might be more as well)

Also, if finding an existing plugin to do this doesn't work out, what about integrating with the maven settings builder?
https://github.com/apache/maven/blob/a2b800de32cdb9adc1e64a43a0fc32e3ba878103/maven-core/src/main/java/org/apache/maven/settings/MavenSettingsBuilder.java#L42

Reply:
@lukecwik the gradle-maven-publish-auth plugin looks like a good fit, but AFAIK the LGPL license would be in the way of using it (hard dependency in our build system).

Reply:
I see, followed up with one plugin author: https://github.com/sebersole/gradle-maven-publish-auth/issues/16#issuecomment-613040172

Reply:
@lukecwik I found a plugin with Apache license that achieves what is required. PTAL.